src/main/scala/sbtcrossproject/CrossProject.scala

```diff
@@ -1,184 +1,115 @@
-import scala.language.experimental.macros
-
-import scala.reflect.macros.Context
-
-import sbt._
-import Keys._
-import Project.projectToRef
-
 import java.io.File
 
-final class CrossProject private (
-    crossType: CrossType,
-    val projects: Map[CrossPlatform, Project]
-) {
-  import CrossProject._
-
-  // Transformers for inner projects
-
-  def configure(platform: CrossPlatform, transform: Project => Project) =
-    copy(projects = projects + (platform -> transform(projects(platform))))
-
-  def settings(platform: CrossPlatform, ss: Def.Setting[_]*): CrossProject =
-    configure(platform, _.settings(ss: _*))
-
-  // Scala.js-style cross project aliases
-
-  /** Project for JVM platform. */
-  def jvm: Project = projects(JVM)
-
-  /** Transform the underlying JVM project */
-  def jvmConfigure(transform: Project => Project): CrossProject =
-    configure(JVM, transform)
+import sbt.{AutoPlugin, Configuration, Plugins, Project, ProjectReference}
 
-  /** Add settings specific to the underlying JVM project */
-  def jvmSettings(ss: Def.Setting[_]*): CrossProject =
-    settings(JVM, ss: _*)
-
-  // Concrete alteration members
-
-  def aggregate(refs: CrossProject*): CrossProject =
-    mapCrossPlatforms { p =>
-      projects(p).aggregate(refs.map(_.projects(p): ProjectReference): _*)
-    }
-
-  def configs(cs: Configuration*): CrossProject =
-    mapProjects(_.configs(cs: _*))
+import scala.language.experimental.macros
+import scala.language.implicitConversions
+import scala.reflect.macros.Context
 
-  def configureCross(transforms: (CrossProject => CrossProject)*): CrossProject =
-    transforms.foldLeft(this)((p, t) => t(p))
+/**
+ * Created by grinder on 17.07.16.
+ */
+class CrossProject[A <: Platform] private (
+    id: String,
+    root: File,
+    val projects: Map[Platform, Project]) {
+  def settings(t: Platform,
+               settings: Seq[_root_.sbt.Def.Setting[_]]): CrossProject[A] =
+    copy(projects + (t -> projects(t).settings(settings)))
+
+  def path(t: Platform, path: String): CrossProject[A] =
+    copy(projects + (t -> projects(t).in(new File(s"$root/$path"))))
+
+  def platform[U <: Platform](t: U): CrossProject[A with U] = {
+    copy(
+        projects + (t -> Project(s"${id}_${t.name}",
+                                 new File(s"$root/${t.path}"))))
+  }
 
-  def configureAll(transforms: (Project => Project)*): CrossProject =
-    mapProjects(_.configure(transforms: _*))
+  def aggregate[U <: Platform](
+      refs: CrossProject[U]*): CrossProject[A with U] = {
+    copy[A with U](
+        refs
+          .map(_.projects)
+          .foldLeft(projects.map(x => x._1 -> List(x._2)))((x, y) =>
+                x ++ y.map({
+          case (k, v) => k -> (if (x.get(k).isDefined) x(k) :+ v else List(v))
+        }))
+          .map(x =>
+                x._1 -> (x._2 match {
+              case head :: tail =>
+                head.aggregate(tail.map(x => x: ProjectReference): _*)
+            })))
```

Comment:
In fact this code 
1) finding platform intersection between this and param
2) call aggregate for each found platform

And yes I want to refactor this code.
Could you confirm that I understand goals for this method correctly?